tests/obj_factory_test.cpp

```diff
@@ -101,3 +101,82 @@ BOOST_AUTO_TEST_CASE(non_copyable_obj_factory) {
     BOOST_CHECK_EQUAL(*(obj_factory["OpenData"]->val), 4);
     BOOST_CHECK_EQUAL(obj_factory.size(), 6);
 }
+
+BOOST_AUTO_TEST_CASE(erase_menber_into_obj_factory) {
+
+    // Erase with uri key
+    {
+        for (size_t i = 0; i < 4; ++i) {
+
+            navitia::ObjFactory<HeaderInt> obj_factory;
+
+            std::array<std::string, 4> uris = {"Val_1", "Val_2", "Val_3", "Val_4"};
+            std::array<size_t, 4> values = {0, 1, 2, 3};
+
+            // Data
+            for (size_t j = 0; j < 4; ++j) {
+                auto data = HeaderInt();
+                data.val = values[j];
+                obj_factory.insert(uris[j], std::move(data));
+            }
+            BOOST_CHECK_EQUAL(obj_factory.size(), 4);
+            for (size_t j = 0; j < obj_factory.size(); ++j) {
+                BOOST_CHECK_EQUAL((obj_factory[uris[i]]->val), i);
+            }
+
+            // Erase with Idx
+            // Be carefull, erase function shifts (<<) elems inside vector
+            // It changes the idx for all shifted elems
+            BOOST_CHECK_EQUAL(obj_factory.erase("Val_10"), false);
+            BOOST_CHECK_EQUAL(obj_factory.erase(uris[i]), true);
+            BOOST_CHECK_EQUAL(obj_factory.size(), 3);
+            BOOST_CHECK_EQUAL(obj_factory.if_exist(uris[i]), false);
+            for (size_t j = 0; j < obj_factory.size(); ++j) {
+                if (j < i) {
+                    BOOST_CHECK_EQUAL(obj_factory.get_mut(navitia::Idx<HeaderInt>(j))->val, j);
+                }
+                else {
+                    BOOST_CHECK_EQUAL(obj_factory.get_mut(navitia::Idx<HeaderInt>(j))->val, j + 1);
```

Comment:
So `obj_factory[uris[j]]->val.uri != uris[j]`, if I understand correctly.
How is that what we want? To me it's inconsistent and map doesn't point to the good elem of vec.
Did I miss something?

Reply:
That test all erase for each position when we have 4 elements (with the main loop).

Erase in position 0 : 
[0, 1, 2, 3] -> [X, 1, 2, 3] -> after shift  [1, 2, 3]

Erase in position 1 : 
[0, 1, 2, 3] -> [0, X, 2, 3] -> after shift  [0, 2, 3]

Erase in position 2 : 
[0, 1, 2, 3] -> [0, 1, X, 3] -> after shift  [0, 1, 3]

Erase in position 3 : 
[0, 1, 2, 3] -> [0, 1, 2, X] -> after shift  [0, 1, 2]

Reply:
Yep, but as object are stored directly in vec, if you erase in position 0, then map[1] points to actual object 2 in vector, no?
To me indexes in map are not updated after erase in vec.
So next time you want to get the object by map you get the wrong object.

Reply:
Ah it's unique_ptr, OK it's fine.